DataCollector.py

```diff
@@ -70,16 +69,15 @@ def run(self):
 
     def setUname(self, uname):
         with self.dataLock:
-            if self.filename:
-                self._writeToFile()
 
-            self.filename =  os.path.join(
-                                self.log_dir,
+            os.makedirs(self.LOG_DIR, exist_ok=True)
+            self.filename = os.path.join(
+                                self.LOG_DIR,
```

Comment:
I de t is wrong, I’ll show you

Reply:
```
self.filename = os.path.join(
    self.LOG_DIR,
    '{}_{}.json'.format(uname,time.strftime("%y%m%d%H%M%S", time.localtime()))
)
```

indentation is 4 spaces, have the element close at the same level as the line which had the elemt open